tests/src/Interop/COM/NETClients/Primitives/NumericTests.cs

```diff
@@ -200,5 +201,39 @@ private void Marshal_ManyInts()
             Console.WriteLine($"{expected.GetType().Name} 12 test invariant: 1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10 + 11 + 12 = {expected}");
             Assert.IsTrue(expected == this.server.Add_ManyInts12(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12));
         }
+
+        private void Marshal_Struct_Return()
```

Comment:
Does this bug also exist for a `DllImport(CallingConvention.ThisCall)`, is it worth explicitly testing?
* More specifically wondering about the case where a user manually declares a COM VTBL and marshals the function pointer (or, eventually, when using the function pointer feature in C#).

Reply:
I'll take a look.  I have a feeling that the bug will also exist for `ThisCall` and that I'll need a slightly more involved fix for that.

Reply:
I've added in a test case and a fix for `thiscall` functions.